rmf_robot_sim_gz_plugins/src/TeleportDispenser.cpp

```diff
@@ -101,6 +127,186 @@ TeleportDispenserPlugin::~TeleportDispenserPlugin()
   rclcpp::shutdown();
 }
 
+void TeleportDispenserPlugin::init_ros_node()
+{
+  if (!_ros_node)
+  {
+    RCLCPP_ERROR(_ros_node->get_logger(),
+      "No ROS node created for TeleportDispenser plugin!");
+    return;
+  }
+
+  _fleet_state_sub = _ros_node->create_subscription<FleetState>(
+    "/fleet_states",
+    rclcpp::SystemDefaultsQoS().keep_last(10),
+    std::bind(&TeleportDispenserPlugin::fleet_state_cb, this,
+    std::placeholders::_1));
+
+  _state_pub = _ros_node->create_publisher<DispenserState>(
+    "/dispenser_states", 10);
+
+  _request_sub = _ros_node->create_subscription<DispenserRequest>(
+    "/dispenser_requests",
+    rclcpp::SystemDefaultsQoS().keep_last(10).reliable(),
+    std::bind(&TeleportDispenserPlugin::dispenser_request_cb, this,
+    std::placeholders::_1));
+
+  _result_pub = _ros_node->create_publisher<DispenserResult>(
+    "/dispenser_results", 10);
+
+  _current_state.guid = _guid;
+  _current_state.mode = DispenserState::IDLE;
+}
+
+void TeleportDispenserPlugin::send_dispenser_response(uint8_t status) const
+{
+  auto response = make_response<DispenserResult>(
+    status, _sim_time, _latest.request_guid, _guid);
```

Comment:
This function is defined in `rmf_robot_sim_common` but it doesn't really need to be there anymore now. I'm a bit on the fence on whether to keep it as a template and reduce duplication between `Dispensers` and `Ingestors` or just duplicate its implementation in the two plugins